pilot/pkg/model/telemetry.go

```diff
@@ -933,6 +1006,125 @@ var waypointStatsConfig = protoconv.MessageToAny(&udpa.TypedStruct{
 // STOP. DO NOT UPDATE THIS WITHOUT UPDATING buildHTTPTelemetryFilter and buildTCPTelemetryFilter.
 const telemetryFilterHandled = 14
 
+func ECDSResourceName(protocol networking.ListenerProtocol, class networking.ListenerClass, nodeType NodeType) string {
+	p := "HTTP"
+	if protocol == networking.ListenerProtocolTCP {
+		p = "TCP"
+	}
+
+	metricClass := "client"
+	if class == networking.ListenerClassSidecarInbound {
+		metricClass = "server"
+	}
+
+	node := ""
+	if nodeType == Waypoint {
+		node = "waypoint"
+	}
+
+	return fmt.Sprintf("stats|%s|%s|%s", p, metricClass, node)
+}
+
+func buildHTTPTypedExtensionConfig(class networking.ListenerClass, metricsCfg []telemetryFilterConfig) []*core.TypedExtensionConfig {
+	res := make([]*core.TypedExtensionConfig, 0, telemetryFilterHandled)
+	for _, cfg := range metricsCfg {
+		switch cfg.Provider.GetProvider().(type) {
+		case *meshconfig.MeshConfig_ExtensionProvider_Prometheus:
+			if cfg.NodeType == Waypoint {
+				res = append(res, &core.TypedExtensionConfig{
+					Name: ECDSResourceName(networking.ListenerProtocolHTTP, class, cfg.NodeType),
+					TypedConfig: protoconv.MessageToAny(&hcm.HttpFilter{
+						Name:       xds.StatsFilterName,
+						ConfigType: &hcm.HttpFilter_TypedConfig{TypedConfig: waypointStatsConfig},
+					}),
+				})
+			} else {
+				if statsCfg := generateStatsConfig(class, cfg); statsCfg != nil {
+					res = append(res, &core.TypedExtensionConfig{
+						Name: ECDSResourceName(networking.ListenerProtocolHTTP, class, cfg.NodeType),
```

Comment:
You can have SD and prometheus; the key doesn't take this into account.

Reply:
This PR is still pretty rough, I remember adding it at the beginning.
I will complete these details if there is no objection to this direction.